pkg/builder/build_test.go

```diff
@@ -206,6 +209,104 @@ var _ = Describe("application", func() {
 			close(done)
 		}, 10)
 	})
+
+	Describe("Start with Controller", func() {
+		depName := "deploy-name-2"
+		It("should Reconcile objects", func(done Done) {
+			By("Creating the application")
```

Comment:
these tests effectively do the same thing after setting up the controller.  We should prob refactor out into a help above for easier maintenance

Reply:
or invert and do something like

```go
func DescribeAController(setupTheController func() manager.Manager) {
   It("should reconcile objects", func(done Done) {
       mgr := setupTheController()
       // ...
   })
}
```

Reply:
Done